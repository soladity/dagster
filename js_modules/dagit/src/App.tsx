import * as React from "react";

import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";

import CustomAlertProvider from "./CustomAlertProvider";
import { NonIdealState } from "@blueprintjs/core";
import { PipelineExecutionRoot } from "./execute/PipelineExecutionRoot";
import { PipelineExecutionSetupRoot } from "./execute/PipelineExecutionSetupRoot";
import PipelineExplorerRoot from "./PipelineExplorerRoot";
import PythonErrorInfo from "./PythonErrorInfo";
import { RootPipelinesQuery } from "./types/RootPipelinesQuery";
import { RunRoot } from "./runs/RunRoot";
import { RunsRoot } from "./runs/RunsRoot";
import { SolidsRoot } from "./solids/SolidsRoot";
import SchedulesRoot from "./schedules/SchedulesRoot";
import { ScheduleRoot } from "./schedules/ScheduleRoot";
import { TopNav } from "./TopNav";
import gql from "graphql-tag";
import { useQuery } from "react-apollo";

function extractData(result?: RootPipelinesQuery) {
  if (!result || !result.pipelinesOrError) {
    return { pipelines: [], error: null };
  }
  if (result.pipelinesOrError.__typename === "PipelineConnection") {
    return { pipelines: result.pipelinesOrError.nodes, error: null };
  } else {
    return { pipelines: [], error: result.pipelinesOrError };
  }
}

const AppRoutes = () => (
  <Switch>
    <Route path="/runs/:runId" component={RunRoot} />
    <Route path="/runs" component={RunsRoot} exact={true} />
    <Route path="/solids/:name?" component={SolidsRoot} />
    <Route path="/p/:pipelineName/runs/:runId" component={RunRoot} />
    <Redirect
      from="/p/:pipelineName"
      exact={true}
      to="/p/:pipelineName/explore"
    />

    <Route
      path="/p/:pipelineName/execute/setup"
      component={PipelineExecutionSetupRoot}
    />
    <Route path="/p/:pipelineName/execute" component={PipelineExecutionRoot} />
    {/* Capture solid subpath in a regex match */}
    <Route
      path="/p/:pipelineName/explore(/?.*)"
      component={PipelineExplorerRoot}
    />
    {/* Legacy redirects */}
    <Redirect
      from="/execute/:pipelineName/setup"
      to="/p/:pipelineName/execute/setup"
    />
    <Redirect from="/execute/:pipelineName" to="/p/:pipelineName/execute" />
    <Redirect
      from="/explore/:pipelineName/:rest?"
      to="/p/:pipelineName/explore/:rest?"
    />
    {/* Index default */}
    <Route path="/scheduler/:rest?" component={SchedulesRoot} />
    <Route path="/schedule/:scheduleName" component={ScheduleRoot} />
    <Route
      render={() => (
        <NonIdealState
          title="No pipeline selected"
          description="Select a pipeline in the navbar"
        />
      )}
    />
  </Switch>
);

export const App: React.FunctionComponent = () => {
  const result = useQuery<RootPipelinesQuery>(ROOT_PIPELINES_QUERY, {
    fetchPolicy: "cache-and-network"
  });
  const { pipelines, error } = extractData(result.data);

  return (
    <BrowserRouter>
      <TopNav pipelines={pipelines} />
      {error ? (
        <PythonErrorInfo
          contextMsg={`${error.__typename} encountered when loading pipelines:`}
          error={error}
          centered={true}
        />
      ) : (
        <AppRoutes />
      )}
      <CustomAlertProvider />
    </BrowserRouter>
  );
};

export const ROOT_PIPELINES_QUERY = gql`
  query RootPipelinesQuery {
    pipelinesOrError {
      __typename
      ... on PythonError {
        message
        stack
      }
      ... on PipelineConnection {
        nodes {
          ...TopNavPipelinesFragment
        }
      }
    }
  }

  ${TopNav.fragments.TopNavPipelinesFragment}
`;
