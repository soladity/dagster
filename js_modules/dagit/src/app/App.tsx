import * as React from 'react';
import {BrowserRouter, Route, Switch} from 'react-router-dom';

import {CustomAlertProvider} from 'src/app/CustomAlertProvider';
import {CustomConfirmationProvider} from 'src/app/CustomConfirmationProvider';
import {CustomTooltipProvider} from 'src/app/CustomTooltipProvider';
import {APP_PATH_PREFIX} from 'src/app/DomUtils';
import {FallthroughRoot} from 'src/app/FallthroughRoot';
import {FeatureFlagsRoot} from 'src/app/FeatureFlagsRoot';
import {SettingsRoot} from 'src/app/SettingsRoot';
import {TimezoneProvider} from 'src/app/time/TimezoneContext';
import {useDocumentTitle} from 'src/hooks/useDocumentTitle';
import {InstanceRoot} from 'src/instance/InstanceRoot';
import {LeftNavBranch} from 'src/nav/LeftNavBranch';
import {AllSchedulesRoot} from 'src/schedules/AllSchedulesRoot';
import {AllSensorsRoot} from 'src/sensors/AllSensorsRoot';
import {WorkspaceProvider} from 'src/workspace/WorkspaceContext';
import {WorkspaceRoot} from 'src/workspace/WorkspaceRoot';

export const AppContent = () => {
  useDocumentTitle('Dagit');
  return (
    <div style={{display: 'flex', height: '100%'}}>
      <WorkspaceProvider>
        <LeftNavBranch />
        <CustomConfirmationProvider>
          <Switch>
            <Route path="/flags" component={FeatureFlagsRoot} />
            <Route path="/instance" component={InstanceRoot} />
            <Route path="/workspace" component={WorkspaceRoot} />
            <Route path="/schedules" component={AllSchedulesRoot} />
            <Route path="/sensors" component={AllSensorsRoot} />
            <Route path="/settings" component={SettingsRoot} />
            <Route path="*" component={FallthroughRoot} />
          </Switch>
          <CustomTooltipProvider />
          <CustomAlertProvider />
        </CustomConfirmationProvider>
      </WorkspaceProvider>
    </div>
  );
};

export const App = () => (
  <BrowserRouter basename={APP_PATH_PREFIX}>
    <TimezoneProvider>
      <AppContent />
    </TimezoneProvider>
  </BrowserRouter>
);
