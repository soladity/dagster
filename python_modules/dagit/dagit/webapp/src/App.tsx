import * as React from "react";
import { Switch, Route } from "react-router";
import { BrowserRouter } from "react-router-dom";
import Page from "./Page";
import PipelinesContainer from "./PipelinesContainer";

export default class App extends React.Component {
  public render() {
    return (
      <Page>
        <BrowserRouter>
          <Route
            path="/:pipeline?"
            render={({ match, history }) => {
              return (
                <PipelinesContainer
                  pipelineName={match.params.pipeline || ""}
                  history={history}
                />
              );
            }}
          />
        </BrowserRouter>
      </Page>
    );
  }
}
