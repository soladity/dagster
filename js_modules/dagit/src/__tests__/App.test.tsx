import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { App } from "../App";
import AppCache from "../AppCache";
import { MOCKS } from "./AppMocks";
import { MockedProvider } from "./MockedProvider";

function createNodeMock(element: any) {
  if (element.type === "div") {
    return {
      querySelector() {
        return null;
      },
      querySelectorAll() {
        return [];
      },
      getBoundingClientRect() {
        return {
          left: 0,
          top: 0,
          right: 1000,
          bottom: 1000,
          x: 0,
          y: 0,
          width: 1000,
          height: 1000
        };
      }
    };
  }
  return null;
}

async function testApp() {
  const component = TestRenderer.create(
    <MockedProvider mocks={MOCKS} addTypename={true} cache={AppCache}>
      <App />
    </MockedProvider>,
    { createNodeMock }
  );

  // wait for the graphql promises to settle. Usually one tick is enough for us,
  // except when a query resolves and triggers another query lower in the tree.
  await new Promise(resolve => setTimeout(resolve, 1000));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}

it("renders without error", async () => {
  await testApp();
});

it("renders pipeline page", async () => {
  beforeEach(() => {
    window.history.pushState({}, "", "/p/airline_demo_ingest_pipeline/explore");
  });
  await testApp();
});

it("renders pipeline solid page", async () => {
  beforeEach(() => {
    window.history.pushState(
      {},
      "",
      "/p/airline_demo_ingest_pipeline/explore/download_q2_sfo_weather"
    );
  });
  await testApp();
});

it("renders type page", async () => {
  beforeEach(() => {
    window.history.pushState(
      {},
      "",
      "/p/airline_demo_ingest_pipeline/explore/download_q2_sfo_weather?types=true"
    );
  });
  await testApp();
});

it("renders type page", async () => {
  beforeEach(() => {
    window.history.pushState(
      {},
      "",
      "/p/airline_demo_ingest_pipeline/explore/download_q2_sfo_weather?typeExplorer=PandasDataFrame"
    );
  });
  await testApp();
});

it("renders solids explorer", async () => {
  beforeEach(() => {
    window.history.pushState({}, "", "/solids/s3_to_df");
  });
  await testApp();
});

it("renders execution", async () => {
  beforeEach(() => {
    window.history.pushState({}, "", "/p/airline_demo_ingest_pipeline/execute");
  });
  await testApp();
});
