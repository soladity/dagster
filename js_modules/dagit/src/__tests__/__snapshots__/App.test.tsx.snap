// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders execution 1`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minWidth": 230,
                  }
                }
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  Select a pipeline...
                </span>
                <span
                  className="bp3-icon bp3-icon-double-caret-vertical"
                  icon="double-caret-vertical"
                >
                  <svg
                    data-icon="double-caret-vertical"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      double-caret-vertical
                    </desc>
                    <path
                      d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
        </span>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="sc-dliRfk bAstHC"
        href="/runs"
        onClick={[Function]}
      >
        Runs
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/scheduler"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        className="sc-dliRfk bAstHC active"
        href="/solids"
        onClick={[Function]}
      >
        Solids
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-bwzfXH iUsTio"
        style={
          Object {
            "background": "#3DCC91",
          }
        }
        title="Connecting..."
      />
      <span />
    </div>
  </div>,
  <div
    className="sc-cIShpX SyZHX"
  >
    <div
      className="sc-kafWEX cQLFPf"
    >
      <div
        className="bp3-non-ideal-state"
      >
        <div
          className="bp3-non-ideal-state-visual"
        >
          <span
            className="bp3-icon bp3-icon-error"
            icon="error"
          >
            <svg
              data-icon="error"
              height={60}
              viewBox="0 0 20 20"
              width={60}
            >
              <desc>
                error
              </desc>
              <path
                d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
        <h4
          className="bp3-heading"
        >
          GraphQL Error - see console for details
        </h4>
      </div>
    </div>
  </div>,
]
`;

exports[`renders pipeline page 1`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minWidth": 230,
                  }
                }
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  Select a pipeline...
                </span>
                <span
                  className="bp3-icon bp3-icon-double-caret-vertical"
                  icon="double-caret-vertical"
                >
                  <svg
                    data-icon="double-caret-vertical"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      double-caret-vertical
                    </desc>
                    <path
                      d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
        </span>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="sc-dliRfk bAstHC"
        href="/runs"
        onClick={[Function]}
      >
        Runs
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/scheduler"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/solids"
        onClick={[Function]}
      >
        Solids
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-bwzfXH iUsTio"
        style={
          Object {
            "background": "#3DCC91",
          }
        }
        title="Connecting..."
      />
      <span />
    </div>
  </div>,
  <div
    className="bp3-non-ideal-state"
  >
    <h4
      className="bp3-heading"
    >
      No pipeline selected
    </h4>
    <div>
      Select a pipeline in the navbar
    </div>
  </div>,
]
`;

exports[`renders pipeline solid page 1`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button bp3-disabled"
                disabled={true}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minWidth": 230,
                  }
                }
                tabIndex={-1}
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  airline_demo_ingest_pipeline
                </span>
                <span
                  className="bp3-icon bp3-icon-double-caret-vertical"
                  icon="double-caret-vertical"
                >
                  <svg
                    data-icon="double-caret-vertical"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      double-caret-vertical
                    </desc>
                    <path
                      d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
        </span>
        <div
          className="bp3-navbar-divider"
        />
        <div
          className="sc-cqCuEk jLQGya"
        >
          <a
            className="sc-dliRfk bAstHC active"
            href="/p/airline_demo_ingest_pipeline/explore"
            onClick={[Function]}
          >
            Explore
          </a>
          <a
            className="sc-dliRfk bAstHC"
            href="/p/airline_demo_ingest_pipeline/execute"
            onClick={[Function]}
          >
            Execute
          </a>
        </div>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="sc-dliRfk bAstHC"
        href="/runs"
        onClick={[Function]}
      >
        Runs
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/scheduler"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/solids"
        onClick={[Function]}
      >
        Solids
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-bwzfXH iUsTio"
        style={
          Object {
            "background": "#3DCC91",
          }
        }
        title="Connecting..."
      />
      <span />
    </div>
  </div>,
  <div
    className="sc-cIShpX SyZHX"
  >
    <div
      className="sc-kafWEX cQLFPf"
    >
      <div
        className="bp3-non-ideal-state"
      >
        <div
          className="bp3-non-ideal-state-visual"
        >
          <span
            className="bp3-icon bp3-icon-error"
            icon="error"
          >
            <svg
              data-icon="error"
              height={60}
              viewBox="0 0 20 20"
              width={60}
            >
              <desc>
                error
              </desc>
              <path
                d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
        <h4
          className="bp3-heading"
        >
          GraphQL Error - see console for details
        </h4>
      </div>
    </div>
  </div>,
]
`;

exports[`renders solids explorer 1`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button bp3-disabled"
                disabled={true}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minWidth": 230,
                  }
                }
                tabIndex={-1}
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  airline_demo_ingest_pipeline
                </span>
                <span
                  className="bp3-icon bp3-icon-double-caret-vertical"
                  icon="double-caret-vertical"
                >
                  <svg
                    data-icon="double-caret-vertical"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      double-caret-vertical
                    </desc>
                    <path
                      d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
        </span>
        <div
          className="bp3-navbar-divider"
        />
        <div
          className="sc-cqCuEk jLQGya"
        >
          <a
            className="sc-dliRfk bAstHC active"
            href="/p/airline_demo_ingest_pipeline/explore"
            onClick={[Function]}
          >
            Explore
          </a>
          <a
            className="sc-dliRfk bAstHC"
            href="/p/airline_demo_ingest_pipeline/execute"
            onClick={[Function]}
          >
            Execute
          </a>
        </div>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="sc-dliRfk bAstHC"
        href="/runs"
        onClick={[Function]}
      >
        Runs
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/scheduler"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/solids"
        onClick={[Function]}
      >
        Solids
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-bwzfXH iUsTio"
        style={
          Object {
            "background": "#3DCC91",
          }
        }
        title="Connecting..."
      />
      <span />
    </div>
  </div>,
  <div
    className="sc-cIShpX SyZHX"
  >
    <div
      className="sc-kafWEX cQLFPf"
    >
      <div
        className="bp3-non-ideal-state"
      >
        <div
          className="bp3-non-ideal-state-visual"
        >
          <span
            className="bp3-icon bp3-icon-error"
            icon="error"
          >
            <svg
              data-icon="error"
              height={60}
              viewBox="0 0 20 20"
              width={60}
            >
              <desc>
                error
              </desc>
              <path
                d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
        <h4
          className="bp3-heading"
        >
          GraphQL Error - see console for details
        </h4>
      </div>
    </div>
  </div>,
]
`;

exports[`renders type page 1`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button bp3-disabled"
                disabled={true}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minWidth": 230,
                  }
                }
                tabIndex={-1}
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  airline_demo_ingest_pipeline
                </span>
                <span
                  className="bp3-icon bp3-icon-double-caret-vertical"
                  icon="double-caret-vertical"
                >
                  <svg
                    data-icon="double-caret-vertical"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      double-caret-vertical
                    </desc>
                    <path
                      d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
        </span>
        <div
          className="bp3-navbar-divider"
        />
        <div
          className="sc-cqCuEk jLQGya"
        >
          <a
            className="sc-dliRfk bAstHC active"
            href="/p/airline_demo_ingest_pipeline/explore"
            onClick={[Function]}
          >
            Explore
          </a>
          <a
            className="sc-dliRfk bAstHC"
            href="/p/airline_demo_ingest_pipeline/execute"
            onClick={[Function]}
          >
            Execute
          </a>
        </div>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="sc-dliRfk bAstHC"
        href="/runs"
        onClick={[Function]}
      >
        Runs
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/scheduler"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/solids"
        onClick={[Function]}
      >
        Solids
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-bwzfXH iUsTio"
        style={
          Object {
            "background": "#3DCC91",
          }
        }
        title="Connecting..."
      />
      <span />
    </div>
  </div>,
  <div
    className="sc-cIShpX SyZHX"
  >
    <div
      className="sc-kafWEX cQLFPf"
    >
      <div
        className="bp3-non-ideal-state"
      >
        <div
          className="bp3-non-ideal-state-visual"
        >
          <span
            className="bp3-icon bp3-icon-error"
            icon="error"
          >
            <svg
              data-icon="error"
              height={60}
              viewBox="0 0 20 20"
              width={60}
            >
              <desc>
                error
              </desc>
              <path
                d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
        <h4
          className="bp3-heading"
        >
          GraphQL Error - see console for details
        </h4>
      </div>
    </div>
  </div>,
]
`;

exports[`renders type page 2`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button bp3-disabled"
                disabled={true}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minWidth": 230,
                  }
                }
                tabIndex={-1}
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  airline_demo_ingest_pipeline
                </span>
                <span
                  className="bp3-icon bp3-icon-double-caret-vertical"
                  icon="double-caret-vertical"
                >
                  <svg
                    data-icon="double-caret-vertical"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      double-caret-vertical
                    </desc>
                    <path
                      d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
        </span>
        <div
          className="bp3-navbar-divider"
        />
        <div
          className="sc-cqCuEk jLQGya"
        >
          <a
            className="sc-dliRfk bAstHC active"
            href="/p/airline_demo_ingest_pipeline/explore"
            onClick={[Function]}
          >
            Explore
          </a>
          <a
            className="sc-dliRfk bAstHC"
            href="/p/airline_demo_ingest_pipeline/execute"
            onClick={[Function]}
          >
            Execute
          </a>
        </div>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="sc-dliRfk bAstHC"
        href="/runs"
        onClick={[Function]}
      >
        Runs
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/scheduler"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/solids"
        onClick={[Function]}
      >
        Solids
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-bwzfXH iUsTio"
        style={
          Object {
            "background": "#3DCC91",
          }
        }
        title="Connecting..."
      />
      <span />
    </div>
  </div>,
  <div
    className="sc-cIShpX SyZHX"
  >
    <div
      className="sc-kafWEX cQLFPf"
    >
      <div
        className="bp3-non-ideal-state"
      >
        <div
          className="bp3-non-ideal-state-visual"
        >
          <span
            className="bp3-icon bp3-icon-error"
            icon="error"
          >
            <svg
              data-icon="error"
              height={60}
              viewBox="0 0 20 20"
              width={60}
            >
              <desc>
                error
              </desc>
              <path
                d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
        <h4
          className="bp3-heading"
        >
          GraphQL Error - see console for details
        </h4>
      </div>
    </div>
  </div>,
]
`;

exports[`renders without error 1`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minWidth": 230,
                  }
                }
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  Select a pipeline...
                </span>
                <span
                  className="bp3-icon bp3-icon-double-caret-vertical"
                  icon="double-caret-vertical"
                >
                  <svg
                    data-icon="double-caret-vertical"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      double-caret-vertical
                    </desc>
                    <path
                      d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                      fillRule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
        </span>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="sc-dliRfk bAstHC"
        href="/runs"
        onClick={[Function]}
      >
        Runs
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/scheduler"
        onClick={[Function]}
      >
        Schedule
      </a>
      <a
        className="sc-dliRfk bAstHC"
        href="/solids"
        onClick={[Function]}
      >
        Solids
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-bwzfXH iUsTio"
        style={
          Object {
            "background": "#3DCC91",
          }
        }
        title="Connecting..."
      />
      <span />
    </div>
  </div>,
  <div
    className="bp3-non-ideal-state"
  >
    <h4
      className="bp3-heading"
    >
      No pipeline selected
    </h4>
    <div>
      Select a pipeline in the navbar
    </div>
  </div>,
]
`;
